### PR TITLE
Sidekick enhancements

### LINKED ISSUE #783: Sidekick: do not reuse opened windows
The sidekick is trying to reuse windows it opened to avoid cluttering the browser tabs, but this is more troublesome than helpful in real life scenarios. It should rather just open a new window every time.

cc @davidnuescheler

### LINKED ISSUE #818: Sidekick: use .lnk URLs instead of direct action calls
See https://github.com/adobe/helix-content-proxy/issues/328

### LINKED ISSUE #848: Sidekick: always reload context when activated
The sidekick bookmarklet currently only loads the config and location once per window. This can lead to inconsistent behavior in conjunction with browsing different markdown files in OneDrive, or when calling the sidekick on different projects.

### LINKED ISSUE #854: Sidekick: use HLXPURGE instead of helix-purge
The sidekick should no longer use runtime actions directly.

Relates to https://github.com/adobe/helix-purge/issues/197

### LINKED ISSUE #859: Sidekick: use main as default branch instead of master

### PR DESCRIPTION
**Pull request by rofe**
Fix #848 
Fix #818 
Fix #783 
Fix #859 
Fix #854

### CODE REVIEW

File: tools/sidekick/app.js [hunk 1]
```diff
@@ -331,23 +348,21 @@
...
           if (sk.isEditor()) {
-            url = new URL('https://adobeioruntime.net/api/v1/web/helix/helix-services/content-proxy@v2');
-            url.search = new URLSearchParams([
-              ['owner', config.owner],
-              ['repo', config.repo],
-              ['ref', config.ref || 'main'],
-              ['path', '/'],
-              ['lookup', location.href],
-            ]).toString();
+            const previewPath = `/hlx_${btoa(location.href).replace(/\+/, '-').replace(/\//, '_')}.lnk`;
```
Comment (kptdobe) [inline tools/sidekick/app.js:355]: What are the 2 `replace(...)` doing on the b64 encoded href ?

**Review by kptdobe — APPROVED** [review]

File: tools/sidekick/app.js (hunk 1 above)
Comment (rofe) [reply tools/sidekick/app.js:355]: They're to keep the b64 safe for use in a URL.

Comment (kptdobe) [reply tools/sidekick/app.js:355]: ah! I did not know you can get a `+` or a `/` in b64 :)

Comment (rofe) [reply tools/sidekick/app.js:355]: Maybe this is obsolete... tbh I took it over from a NodeJS implementation by @tripodsan 😄

Comment (tripodsan) [reply tools/sidekick/app.js:355]: `btoa` still uses the original set:

> The algorithm used by atob() and btoa() is specified in RFC 4648, section 4.

```
     Value Encoding  Value Encoding  Value Encoding  Value Encoding
         0 A            17 R            34 i            51 z
         1 B            18 S            35 j            52 0
         2 C            19 T            36 k            53 1
         3 D            20 U            37 l            54 2
         4 E            21 V            38 m            55 3
         5 F            22 W            39 n            56 4
         6 G            23 X            40 o            57 5
         7 H            24 Y            41 p            58 6
         8 I            25 Z            42 q            59 7
         9 J            26 a            43 r            60 8
        10 K            27 b            44 s            61 9
        11 L            28 c            45 t            62 +
        12 M            29 d            46 u            63 /
        13 N            30 e            47 v
        14 O            31 f            48 w         (pad) =
        15 P            32 g            49 x
        16 Q            33 h            50 y
```

**Comment by rofe** [comment]
OK I think now it's ready to merge. @kptdobe could you give it another quick look?

**Review by kptdobe — APPROVED** [review]
Just to double-check: I do not see some `main` branch specific new tests while you are fixing #859 (breaking feature). Is it expected ?

**Comment by rofe** [comment]
> Just to double-check: I do not see some `main` branch specific new tests while you are fixing #859 (breaking feature). Is it expected ?

Yes, I had to adjust the tests to specify `master` instead of no branch like before, but I can add a dedicated test for it.

**Comment by rofe** [comment]
@kptdobe I added the test as well as #854 (last feature in this PR, promised! 😉)

**Review by kptdobe — APPROVED** [review]

**Pull request merged**